util/ranger/ranger_test.go

```diff
@@ -660,6 +660,20 @@ func (s *testRangerSuite) TestIndexRangeForUnsignedInt(c *C) {
 			filterConds: "[]",
 			resultStr:   `[(NULL,1) (2,9223372036854775810) (9223372036854775810,+inf]]`,
 		},
+		{
+			indexPos:    0,
+			exprStr:     `a >= -2`,
+			accessConds: "[ge(test.t.a, -2)]",
+			filterConds: "[]",
+			resultStr:   `[[0,+inf]]`,
+		},
+		{
+			indexPos:    0,
+			exprStr:     `a = -2`,
+			accessConds: "[eq(test.t.a, -2)]",
+			filterConds: "[]",
+			resultStr:   `[[0,0]]`,
```

Comment:
This case we should return empty range in fact?

Reply:
So is https://github.com/pingcap/tidb/issues/10447.

Reply:
yes
[[0,0]] is incorrect
I will modify my code